example/hook.go

```diff
@@ -0,0 +1,55 @@
+package example
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Makpoc/gomigen/types"
+)
+
+type ctxKey string
+
+const (
+	startTimeKey ctxKey = "startTime"
+)
+
+// LogHook is an example of a hook that logs messages on entry and exit and the time interval
+// between they are called.
+type LogHook struct {
+	logger *log.Logger
+}
+
+var _ types.Hook = (*LogHook)(nil)
+
+func New(log *log.Logger) *LogHook {
+	return &LogHook{
+		logger: log,
+	}
+}
+
+func (l *LogHook) OnEntry(ctx context.Context, info types.MethodInfo) context.Context {
+	startTime := time.Now()
+	ctx = context.WithValue(ctx, startTimeKey, startTime)
```

Comment:
NIT: You can move this to the return and would save one line :) (Maybe I am nit picking too much)

Reply:
Changed